subprojects/shipkit/src/main/groovy/org/shipkit/internal/gradle/versionupgrade/MergePullRequest.java

```diff
@@ -0,0 +1,94 @@
+package org.shipkit.internal.gradle.versionupgrade;
+
+import java.io.IOException;
+
+import org.gradle.api.GradleException;
+import org.gradle.api.logging.Logger;
+import org.gradle.api.logging.Logging;
+import org.json.simple.JsonArray;
+import org.json.simple.JsonObject;
+import org.json.simple.Jsoner;
+import org.shipkit.internal.gradle.util.BranchUtils;
+import org.shipkit.internal.util.GitHubApi;
+import org.shipkit.internal.util.IncubatingWarning;
+
+class MergePullRequest {
+
+    private static final Logger LOG = Logging.getLogger(MergePullRequest.class);
+
+    public void mergePullRequest(MergePullRequestTask task) throws IOException {
+        mergePullRequest(task, new GitHubApi(task.getGitHubApiUrl(), task.getAuthToken()));
+    }
+
+    public void mergePullRequest(MergePullRequestTask task, GitHubApi gitHubApi) throws IOException {
+        if (task.isDryRun()) {
+            LOG.lifecycle(" Skipping pull request merging due to dryRun = true");
+            return;
+        }
+
+        String headBranch = BranchUtils.getHeadBranch(task.getForkRepositoryName(), task.getVersionBranch());
+
+        IncubatingWarning.warn("merge pull requests");
+        LOG.lifecycle("Waiting for status of a pull request in repository '{}' between base = '{}' and head = '{}'.", task.getUpstreamRepositoryName(), task.getVersionUpgrade().getBaseBranch(), headBranch);
+
+        String sha = retrievePullRequestSha(task, gitHubApi, headBranch);
```

Comment:
Can we get this SHA (along with pull request number) from one of the tasks accessing the pull request (CreatePullRequest and FindOpenPullRequest)? You can use FindOpenPullRequest#provideBranchTo method and sth similar in CreatePullRequest. This way we have one API call less.

We would have to check if it works with both creating a new pull request and with updating the existing one. I'm not sure what happens in Gradle if you depend on a task that is skipped (CreatePullRequest is skipped if we already have an open pull request).

Reply:
It's something worth checking with this potential skipped depdendency, I will do that. I will also try to retrieve sha from previous calls.

Reply:
@wwilk it seems that skipping task don't skip dependant tasks: https://discuss.gradle.org/t/onlyif-affect-only-task-but-not-tasks-that-depends-on/14887 . But I'm looking for confirmation in official docs